src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -262,6 +264,13 @@ namespace cryptonote
   {
     m_blockchain_storage.set_enforce_dns_checkpoints(enforce_dns);
   }
+  //-----------------------------------------------------------------------------------
+  void core::set_txpool_listener(std::weak_ptr<listener::zmq_pub> zmq_pub)
+  {
+    CRITICAL_REGION_LOCAL(m_incoming_tx_lock);
+    m_zmq_pub = std::move(zmq_pub);
+  }
```

Comment:
I can create a new library for `zmq_pub.cpp` with the core and blockchain dependent on that instead. A callback isn't going to work as desired, because the code still needs to be invoked. Those other libs need to know where to jump either way.